pkg/reconciler/utils/configmap_test.go

```diff
@@ -70,23 +71,29 @@ type commonCase struct {
 	wantErr    bool
 }
 
+func testConfigMapBinaryDataEqual(cm1, cm2 *corev1.ConfigMap) bool {
```

Comment:
Doesn't seem to be useful to create an eqFunc and test it. Instead, I think we should test the TargetsConfigMapEqual function.

Reply:
Agreed. But I think it still makes sense to test using a custom eqFunc, so I'll change it to only test the simple always-true or always-false eqFuncs. Also, since the ConfigMap reconciler is technically responsible for handling all kinds of ConfigMaps, not just the broker-targets ConfigMap, then I will add the TargetsConfigMapEqual test in a separate file.

Reply:
Thanks for doing that!

I was wondering if we can add a test to show that the custom eq func returns different result than a deepEqual for two semantically equal protos. For example, one proto with a nil map and the other with an empty map. Instead of a made up one, it's even better if we can reproduce the deepEqual bug in our real reconciliation and prove that the custom func solves that.

Reply:
Makes sense to want to test it, but it seems to be a bit more subtle to recreate than simply comparing a proto with a nil map vs empty map. I'm struggling to recreate it using hand-crafted protos (it might have to do with internal proto state that seems tricky to force from the outside), but I can do it using binary data extracted from my real tests. Testing it with this data would involve hard-coding the binary data in the test file. Do you think this is alright?

Reply:
Didn't realize how hard is it to craft the data. I am fine with nil vs empty map approach. This is good enough to demonstrate the flaw of deepEqual.

Reply:
The problem is that marshalling a proto containing a nil map vs empty map will actually create the same byte representation in my testing. So DeepEqual would be good enough.

After more testing, the problem seems to also be caused by the proto marshaller respecting the ordering of map keys. So, protos containing identical maps with keys listed in different order will have a different binary representation. I'll add this test, since this captures the problem we faced in the issue.

Reply:
Good to know. Thanks for looking into this!